pandas/io/pytables.py

```diff
@@ -4621,7 +4622,7 @@ def _get_converter(kind, encoding):
     if kind == 'datetime64':
         return lambda x: np.asarray(x, dtype='M8[ns]')
     elif kind == 'datetime':
```

Comment:
this branch might not even be hit anymore
can u point to where it is exercised?

Reply:
It'll be reached if IndexCol.convert is called with an IndexCol of kind == 'datetime'.

Reply:
i can read what if says (i wrote this)
what i am asking is which test hits this
this inference should not be true except for an object Index which infers to datetime which we rarely have (as they are always converts)

Reply:
Then I have no idea.

Reply:
you can remove this kind == 'datetime' branch its not hit by anything